projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts

```diff
@@ -333,74 +288,22 @@ export class PreviewsComponent extends AccessibleComponent implements OnInit, On
    * Private method to update the visible previews navigating to the left (previous).
    */
   private previous(): void {
-    if (!this.images) {
+    if (!this.images || !this.previewConfig) {
       throw new Error('Internal library error - images must be defined');
     }
-    // check if prevImage should be blocked
-    if (this.isPreventSliding(0)) {
-      return;
-    }
-
-    if (this.start === 0) {
-      return;
+    if(this.start <= 0) {
+      // check if prevImage should be blocked
+      const preventSliding = !!this.slideConfig && this.slideConfig.infinite === false;
+      if(preventSliding) {
+        return;
+      }
+      this.end = this.images.length;
+    } else {
+      this.end--;
     }
-
-    this.start = Math.max(this.start - 1, 0);
-    this.end = Math.min(this.end - 1, this.images.length);
+    this.start = this.end - Math.min((this.previewConfig.number as number), this.images.length);
 
     this.previews = this.images.filter((img: InternalLibImage, i: number) => i >= this.start && i < this.end);
   }
 
-  /**
-   * Private method to block/permit sliding between previews.
-   * @param number boundaryIndex is the first or the last index of `images` input array
-   * @returns boolean if true block sliding, otherwise not
-   */
```

Comment:
preventSliding depends whether next() or previous() is called, so I moved the isPreventSliding logic within these two methods for better contextualization.